src/gpu_check/copy_canary/copy_canary_cl_buffer.c

```diff
@@ -93,15 +100,25 @@ static void *create_svm_copies(cl_context kern_ctx, cl_command_queue cmd_queue,
             exit(-1);
         }
 
-        char *base_ptr = (char*)m1->handle;
-        char *canary_ptr = base_ptr + (m1->size - POISON_FILL_LENGTH);
-        char *map_ptr = ((char*)svm_canary_copies)+(i*POISON_FILL_LENGTH);
+        char *base_ptr = (char*)m1->main_buff;
+        char *canary_ptr = base_ptr;
+        char *canary_ptr2 = base_ptr + (m1->size + POISON_FILL_LENGTH);
```

Comment:
This seems to be hard-coding that we will always do underflow detection -- if so, could you come up with more descriptive names than just appending "2" to the end?